lib/puppet/provider/templates/webadministration/_getapppools.ps1.erb

```diff
@@ -0,0 +1,3 @@
+# TODO: this is JSON, PS v2 doesnt support it
+# PSCustomObject isnt present in v2
+Get-WebConfiguration -Filter '/system.applicationHost/applicationPools/add' | ConvertTo-Json -Depth 10
```

Comment:
just the provider name is not enough, as all providers will be called "webapplication".